Makefile

```diff
@@ -2,64 +2,70 @@
 
 # Import settings and constants
 include .env
-include constants.mk
 
 SHELL:=/bin/bash
 
 # CONSTANTS
 
-SOLIDITY_VERSION := 0.8.28
+SOLC_VERSION := $(shell cat foundry.toml | grep solc | cut -d= -f2 | xargs echo || echo "0.8.28")
 DEPLOY_SCRIPT := script/Deploy.s.sol:DeployScript
-MULTISIG_MEMBERS_FILE := ./multisig-members.json
+SUPPORTED_VERIFIERS := etherscan blockscout sourcify routescan-mainnet routescan-testnet
 MAKE_TEST_TREE_CMD := deno run ./test/scripts/make-test-tree.ts
-TEST_TREE_MARKDOWN := TEST_TREE.md
+TEST_TREE_MARKDOWN := TESTS.md
 ARTIFACTS_FOLDER := ./artifacts
 LOGS_FOLDER := ./logs
 VERBOSITY := -vvv
 
 NETWORK_NAME:=$(strip $(subst ',, $(subst ",,$(NETWORK_NAME))))
 CHAIN_ID:=$(strip $(subst ',, $(subst ",,$(CHAIN_ID))))
+VERIFIER:=$(strip $(subst ',, $(subst ",,$(VERIFIER))))
 
 TEST_COVERAGE_SRC_FILES := $(wildcard test/*.sol test/**/*.sol src/*.sol src/**/*.sol)
 TEST_SOURCE_FILES := $(wildcard test/*.t.yaml test/integration/*.t.yaml)
 TEST_TREE_FILES := $(TEST_SOURCE_FILES:.t.yaml=.tree)
 DEPLOYMENT_ADDRESS := $(shell cast wallet address --private-key $(DEPLOYMENT_PRIVATE_KEY) 2>/dev/null || echo "NOTE: DEPLOYMENT_PRIVATE_KEY is not properly set on .env" > /dev/stderr)
+MULTISIG_MEMBERS_FILE := ./multisig-members.json
 
 DEPLOYMENT_LOG_FILE=deployment-$(NETWORK_NAME)-$(shell date +"%y-%m-%d-%H-%M").log
 
 # Check values
 
-ifeq ($(filter $(NETWORK_NAME),$(AVAILABLE_NETWORKS)),)
-  $(error Unknown network: $(NETWORK_NAME). Must be one of: $(AVAILABLE_NETWORKS) (see constants.mk))
+ifeq ($(filter $(VERIFIER),$(SUPPORTED_VERIFIERS)),)
+  $(error Unknown verifier: $(VERIFIER). It must be one of: $(SUPPORTED_VERIFIERS))
 endif
 
 # Conditional assignments
 
-ifneq ($(filter $(NETWORK_NAME), $(ETHERSCAN_NETWORKS)),)
+ifeq ($(VERIFIER), etherscan)
 	# VERIFIER_URL := https://api.etherscan.io/api
 	VERIFIER_API_KEY := $(ETHERSCAN_API_KEY)
-	VERIFIER_PARAMS := --etherscan-api-key $(ETHERSCAN_API_KEY)
+	VERIFIER_PARAMS := --verifier $(VERIFIER) --etherscan-api-key $(ETHERSCAN_API_KEY)
 endif
 
-ifneq ($(filter $(NETWORK_NAME), $(BLOCKSCOUT_NETWORKS)),)
+ifeq ($(VERIFIER), blockscout)
 	VERIFIER_URL := https://$(BLOCKSCOUT_HOST_NAME)/api\?
 	VERIFIER_API_KEY := ""
-	VERIFIER_PARAMS = --verifier blockscout --verifier-url "$(VERIFIER_URL)"
+	VERIFIER_PARAMS = --verifier $(VERIFIER) --verifier-url "$(VERIFIER_URL)"
 endif
 
-ifneq ($(filter $(NETWORK_NAME), $(SOURCIFY_NETWORKS)),)
+ifeq ($(VERIFIER), sourcify)
```

Comment:
what happened to sourcify?

Reply:
It's left as a placeholder, no params need to be touched. We could just remove it and be happy as well.